### PR TITLE
feat(route): add xuci again #3485

### PR DESCRIPTION
**Pull request by Insomnia1437**
## Involved Issue / 该 PR 相关 Issue

#3485  (old route is deprecated)

## Example for the Proposed Route(s) / 路由地址示例

```routes
/p-articles/section/works
/p-articles/section/works-poetry
/p-articles/contributors/d
/p-articles/contributors/wcy
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
Change v2 Route Standard to new standard

### CODE REVIEW

File: lib/routes/p-articles/contributors.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        supportScihub: false,
+    },
+    name: '虛詞作者',
+    maintainers: ['Insomnia1437'],
+    handler,
+};
+
+async function handler(ctx) {
+    const author_name: string = ctx.req.param('author');
+    const author_url = new URL(`/contributors/${author_name}`, rootUrl).href;
```
Comment (TonyRL) [inline lib/routes/p-articles/contributors.ts:27]: [`camelCase`](https://docs.rsshub.app/joinus/advanced/script-standard#naming)

File: lib/routes/p-articles/contributors.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        list.map((info) =>
+            cache.tryGet(info.link, async () => {
+                const response = await ofetch(info.link);
+                // const $ = load(response);
+                return ProcessFeed(info, response);
+            })
+        )
+    );
+    return {
+        title: `虚词 p-articles`,
```
Comment (TonyRL) [inline lib/routes/p-articles/contributors.ts:51]:
```suggestion
        title: '虚词 p-articles',
```

File: lib/routes/p-articles/contributors.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        link: new URL(`/contributors/${author_name}`, rootUrl).href,
```
Comment (TonyRL) [inline lib/routes/p-articles/contributors.ts:52]: The value is the same as
https://github.com/DIYgod/RSSHub/blob/961b5e62fa6e9a6eca421b4ba1d5b602ea32a87e/lib/routes/p-articles/contributors.ts#L27
so please reuse the variable decleared

File: lib/routes/p-articles/contributors.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import { Route } from '@/types';
+import ofetch from '@/utils/ofetch';
+import { load } from 'cheerio';
+import cache from '@/utils/cache';
+import { rootUrl, ProcessFeed } from './utils';
+
+export const route: Route = {
+    path: '/contributors/:author',
+    categories: ['reading'],
+    example: '/contributors/黃衍仁',
```
Comment (TonyRL) [inline lib/routes/p-articles/contributors.ts:10]:
```suggestion
    example: '/p-articles/contributors/黃衍仁',
```

File: lib/routes/p-articles/section.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
+import { Route } from '@/types';
+import ofetch from '@/utils/ofetch';
+import { load } from 'cheerio';
+import cache from '@/utils/cache';
+import { rootUrl, ProcessFeed } from './utils';
+
+export const route: Route = {
+    path: '/section/:section',
+    categories: ['reading'],
+    example: '/section/critics',
```
Comment (TonyRL) [inline lib/routes/p-articles/section.ts:10]:
```suggestion
    example: '/p-articles/section/critics',
```

File: lib/routes/p-articles/contributors.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    parameters: { author: '虛詞作者, 可在作者页面 URL 找到' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
```
Comment (TonyRL) [inline lib/routes/p-articles/contributors.ts:19]:
```suggestion
```

File: lib/routes/p-articles/section.ts [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    parameters: { section: '版块名称, 可在对应版块 URL 中找到, 子版块链接用`-`连接' },
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
```
Comment (TonyRL) [inline lib/routes/p-articles/section.ts:19]:
```suggestion
```

File: lib/routes/p-articles/section.ts [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+};
+
+async function handler(ctx) {
+    let section_name: string = ctx.req.param('section');
+    section_name = section_name.replace('-', '/');
+    section_name += '/';
+    const section_url = new URL(section_name, rootUrl).href;
+    const response = await ofetch(section_url);
+    const $ = load(response);
+    const top_info = {
```
Comment (TonyRL) [inline lib/routes/p-articles/section.ts:32]: `camelCase`

File: lib/routes/p-articles/section.ts [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        list.map((info) =>
+            cache.tryGet(info.link, async () => {
+                const response = await ofetch(info.link);
+                // const $ = load(response);
+                return ProcessFeed(info, response);
+            })
+        )
+    );
+    return {
+        title: `虚词 p-articles`,
```
Comment (TonyRL) [inline lib/routes/p-articles/section.ts:58]:
```suggestion
        title: '虚词 p-articles',
```

File: lib/routes/p-articles/section.ts [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        link: new URL(section_name, rootUrl).href,
```
Comment (TonyRL) [inline lib/routes/p-articles/section.ts:59]: Reuse https://github.com/DIYgod/RSSHub/blob/961b5e62fa6e9a6eca421b4ba1d5b602ea32a87e/lib/routes/p-articles/section.ts#L29

File: lib/routes/p-articles/utils.ts [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+const rootUrl: string = 'https://p-articles.com';
+
+const ProcessFeed = (info, data) => {
+    // const $ = cheerio.load(data);
+    const $ = load(data);
+    const author = $('div.detail_title_02 > h4 > a:nth-child(2)').text().trim();
+    info.author = author;
+
+    const date_value = $('div.detail_title_02 > h4 ').text().trim();
```
Comment (TonyRL) [inline lib/routes/p-articles/utils.ts:13]: `camelCase`

**Review by TonyRL — COMMENTED** [review]
Please remove files in https://github.com/DIYgod/RSSHub/tree/master/lib/routes-deprecated/p-articles as well

**Comment by Insomnia1437** [comment]
@TonyRL I changed based on your comment, please review again, thanks

File: lib/routes/p-articles/contributors.ts [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+import { rootUrl, ProcessFeed } from './utils';
+
+export const route: Route = {
+    path: '/contributors/:author',
+    categories: ['reading'],
+    example: '/p-articles/contributors/黃衍仁',
+    parameters: { author: '虛詞作者, 可在作者页面 URL 找到' },
+    name: '虛詞作者',
+    maintainers: ['Insomnia1437'],
+    handler,
```
Comment (TonyRL) [inline lib/routes/p-articles/contributors.ts:14]:
```suggestion
    handler,
    radar: [
        {
            source: ['p-articles.com/contributors/:author'],
        },
    ],
```

File: lib/routes/p-articles/section.ts [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+import { rootUrl, ProcessFeed } from './utils';
+
+export const route: Route = {
+    path: '/section/:section',
+    categories: ['reading'],
+    example: '/p-articles/section/critics',
+    parameters: { section: '版块名称, 可在对应版块 URL 中找到, 子版块链接用`-`连接' },
+    name: '版块',
+    maintainers: ['DW'],
+    handler,
```
Comment (TonyRL) [inline lib/routes/p-articles/section.ts:14]:
```suggestion
    handler,
    radar: [
        {
            source: ['p-articles.com/:section/'],
        },
    ],
```

**Pull request merged**